### PR TITLE
feat: revolve a sketch given an axis and an origin

### LINKED ISSUE #1228: Create a ``revolve_sketch`` API given an axis and an angle
With the capabilities in https://geometry.docs.pyansys.com/version/stable/examples/03_modeling/sweep_chain_profile.html
we should now be capable of implementing on the client side revolve capabilities easily.

### PR DESCRIPTION
**Pull request by RobPasMue**
## Description
Given a sketch, an axis and an origin, we compute the distance between the origin and the sketch plane origin, and revolve the sketch.

## Issue linked
Closes #1228 

## Checklist
- [X] I have tested my changes locally.
- [X] I have added necessary documentation or updated existing documentation.
- [X] I have followed the coding style guidelines of this project.
- [X] I have added appropriate unit tests.
- [X] I have reviewed my changes before submitting this pull request.
- [X] I have linked the issue or issues that are solved to the PR if any.
- [X] I have assigned this PR to myself.
- [X] I have added the minimum version decorator to any new backend method implemented.
- [X] I have made sure that the title of my PR follows [Conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. ``feat: extrude circle to cylinder``)

### CODE REVIEW

File: tests/integration/test_plotter.py [hunk 1]
```diff
@@ -688,3 +690,73 @@ def test_plot_design_point(modeler: Modeler, verify_image_cache):
...
+
+    # Create the circlular profile on the XZ-plane centered at (5, 0, 0)
+    # with a radius of 2
+    plane_profile = Plane(
+        origin=Point3D([path_radius, 0, 0]), direction_x=UNITVECTOR3D_X, direction_y=UNITVECTOR3D_Z
+    )
+    profile = Sketch(plane=plane_profile)
+    profile.circle(Point2D([0, 0]), profile_radius)
+
+    # Revolver the profile around the Z-axis and centered in the absolute origin
```
Comment (PipKat) [inline tests/integration/test_plotter.py:712]: Should this be Revolve?

File: tests/integration/test_design.py [hunk 2]
```diff
@@ -2286,3 +2287,35 @@ def test_create_body_from_loft_profile(modeler: Modeler):
...
+
+    # Create the circlular profile on the XZ-plane centered at (5, 0, 0)
+    # with a radius of 2
+    plane_profile = Plane(
+        origin=Point3D([path_radius, 0, 0]), direction_x=UNITVECTOR3D_X, direction_y=UNITVECTOR3D_Z
+    )
+    profile = Sketch(plane=plane_profile)
+    profile.circle(Point2D([0, 0]), profile_radius)
+
+    # Revolver the profile around the Z-axis and centered in the absolute origin
```
Comment (PipKat) [inline tests/integration/test_design.py:2309]: Should this be Revolve?

Comment (RobPasMue) [reply tests/integration/test_design.py:2309]: Agh damn it. My bad... I will modify it. Good catch @PipKat!

**Comment by RobPasMue** [comment]
@PipKat I added an example - could you give it a quick review whenever you get the chance?

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Modeling: Revolving a sketch
+
+This example shows how use the ``revolve_sketch()`` method to
+revolve a sketch around an axis to create a 3D body. Users can also
+specify the angle of revolution to create a partial body.
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:18]:
```suggestion
This example shows how to use the ``revolve_sketch()`` method to
revolve a sketch around an axis to create a 3D body. You can also
specify the angle of revolution to create a partial body.
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+```{code-cell} ipython3
+# Initialize the modeler for this example notebook
+m = Modeler()
+```
+
+## Example: Creating a quarter of a donut
+
+The following code snippets show how to use the `revolve_sketch()` function to create a
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:42]:
```suggestion
The following code snippets show how to use the ``revolve_sketch()`` function to create a
```
Single backticks render as italics in the doc. You must use double backticks for code entities.

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+quarter of a 3D donut. The process involves defining a quarter of a circle as a profile
+and then revolving it around the Z-axis to create a 3D body.
+
+### Initialize the sketch design
+
+A new design sketch named "quarter-donut" is created.
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:48]:
```suggestion
Create a design sketch named ``quarter-donut``.
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+# Initialize the donut sketch design
+design = m.create_design("quarter-donut")
+```
+
+### Define circle parameters
+
+``path_radius`` is set to 5 units, representing the radius of the circular path that the profile
+circle sweeps along.
+``profile_radius`` is set to 2 units, representing the radius of the profile circle that sweeps
+along the path to create the donut body.
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:60]:
```suggestion
Set ``path_radius``, which represents the radius of the circular path that the profile
circle sweeps along, to ``5`` units.
Set ``profile_radius``, which represents the radius of the profile circle that sweeps
along the path to create the donut body, to ``2`` units.
```
Write in active voice (and use present tense).

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+# Donut parameters
+path_radius = 5
+profile_radius = 2
+```
+
+### Create the profile circle
+
+A circle is created on the XZ-plane centered at the coordinates (5, 0, 0) with a radius
+defined by ``profile_radius``. This circle serves as the profile or cross-sectional
+shape of the donut.
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:72]:
```suggestion
Create a circle on the XZ plane centered at the coordinates ``(5, 0, 0)``
and use``profile_radius`` to define the radius. This circle serves as the
profile or cross-sectional shape of the donut.
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+profile = Sketch(plane=plane_profile)
+profile.circle(Point2D([0, 0]), profile_radius)
+
+profile.plot()
+```
+
+### Perform the revolve operation
+
+The profile circle is revolved around the Z-axis to create a quarter of a donut body.
+The angle of revolution is set to 90 degrees in the default direction, which is counterclockwise.
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:91]:
```suggestion
Revolve te profile circle around the Z axis to create a quarter of a donut body.
Set the angle of revolution to 90 degrees in the default direction, which is counterclockwise.
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+```{code-cell} ipython3
+# Revolve the profile around the Z-axis and centered in the absolute origin
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:94]:
```suggestion
# Revolve the profile around the Z axis and center in the absolute origin
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    sketch=profile,
+    axis=UNITVECTOR3D_Z,
+    angle=Angle(90, unit=UNITS.degrees),
+    rotation_origin=Point3D([0, 0, 0]),
+)
+
+design.plot()
+```
+
+### Performing a revolve operation with a negative angle of revolution
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:107]:
```suggestion
### Perform a revolve operation with a negative angle of revolution
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+A negative angle of revolution can be used to create a quarter of a donut in the opposite direction. The following code snippet shows how to create a quarter of a donut in the clockwise direction. The same profile circle is used, but the angle of revolution is set to -90 degrees.
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:109]:
```suggestion
You can use a negative angle of revolution to create a quarter of a donut in the opposite direction. The following code snippet shows how to create a quarter of a donut in the clockwise direction. The same profile circle is used, but the angle of revolution is set to -90 degrees.
```

File: doc/source/examples/03_modeling/revolving.mystnb [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+```{code-cell} ipython3
+# Initialize the donut sketch design
+design = m.create_design("quarter-donut-negative")
+
+# Revolve the profile around the Z-axis and centered in the absolute origin
```
Comment (PipKat) [inline doc/source/examples/03_modeling/revolving.mystnb:115]:
```suggestion
# Revolve the profile around the Z axis and center in the absolute origin
```

File: doc/source/examples/03_modeling/sweep_chain_profile.mystnb [hunk 13]
```diff
@@ -75,7 +75,7 @@ defined by ``profile_radius``. This circle serves as the profile or cross-sectio
 shape of the donut.
 
 ```{code-cell} ipython3
-# Create the circlular profile on the XZ-plane centered at (5, 0, 0)
+# Create the circular profile on the XZ-plane centered at (5, 0, 0)
```
Comment (PipKat) [inline doc/source/examples/03_modeling/sweep_chain_profile.mystnb:78]:
```suggestion
# Create the circular profile on the XZ plane centered at (5, 0, 0)
```

File: doc/source/examples/03_modeling/sweep_chain_profile.mystnb [hunk 14]
```diff
@@ -90,7 +90,7 @@ Another circle, representing the path along which the profile circle is swept, i
 the XY-plane centered at (0, 0, 0). The radius of this circle is defined by ``path_radius``.
 
 ```{code-cell} ipython3
-# Create the circlular path on the XY-plane centered at (0, 0, 0) with radius 5
+# Create the circular path on the XY-plane centered at (0, 0, 0) with radius 5
```
Comment (PipKat) [inline doc/source/examples/03_modeling/sweep_chain_profile.mystnb:93]:
```suggestion
# Create the circular path on the XY plane centered at (0, 0, 0) with radius 5
```

File: src/ansys/geometry/core/designer/component.py [hunk 15]
```diff
@@ -588,6 +590,76 @@ def sweep_chain(
...
+        rotation_origin: Point3D,
+    ) -> Body:
+        """
+        Create a solid body by revolving a sketch profile around an axis.
+
+        Notes
+        -----
+        It is important that the sketch plane origin is not coincident with the rotation
+        origin. If the sketch plane origin is coincident with the rotation origin, the
+        distance between the two points is zero, and the revolve operation will fail.
```
Comment (PipKat) [inline src/ansys/geometry/core/designer/component.py:610]:
```suggestion
        distance between the two points is zero, and the revolve operation fails.
```

File: tests/integration/test_design.py [hunk 16]
```diff
@@ -2286,3 +2284,60 @@ def test_create_body_from_loft_profile(modeler: Modeler):
...
+def test_revolve_sketch(modeler: Modeler):
+    """Test revolving a circular profile for a quarter donut."""
+    # Initialize the donut sketch design
+    design = modeler.create_design("quarter-donut")
+
+    # Donut parameters
+    path_radius = 5
+    profile_radius = 2
+
+    # Create the circular profile on the XZ-plane centered at (5, 0, 0)
```
Comment (PipKat) [inline tests/integration/test_design.py:2298]:
```suggestion
    # Create the circular profile on the XZ plane centered at (5, 0, 0)
```

File: tests/integration/test_design.py [hunk 17]
```diff
@@ -2286,3 +2284,60 @@ def test_create_body_from_loft_profile(modeler: Modeler):
...
+    # with a radius of 2
+    plane_profile = Plane(
+        origin=Point3D([path_radius, 0, 0]), direction_x=UNITVECTOR3D_X, direction_y=UNITVECTOR3D_Z
+    )
+    profile = Sketch(plane=plane_profile)
+    profile.circle(Point2D([0, 0]), profile_radius)
+
+    # Revolve the profile around the Z-axis and centered in the absolute origin
```
Comment (PipKat) [inline tests/integration/test_design.py:2306]:
```suggestion
    # Revolve the profile around the Z axis and center in the absolute origin
```

File: tests/integration/test_design.py [hunk 18]
```diff
@@ -2286,3 +2284,60 @@ def test_create_body_from_loft_profile(modeler: Modeler):
...
+        rotation_origin=Point3D([0, 0, 0]),
+    )
+
+    assert body.is_surface == False
+    assert body.name == "donut-body"
+    assert np.isclose(body.volume.m, np.pi**2 * 2 * 5, rtol=1e-3)  # quarter of a torus volume
+
+
+def test_revolve_sketch_fail(modeler: Modeler):
+    """Test demonstrating the failure of revolving a sketch when they are located in the
```
Comment (PipKat) [inline tests/integration/test_design.py:2322]: What does the pronoun "they" refer to?

File: tests/integration/test_design.py [hunk 19]
```diff
@@ -2286,3 +2284,60 @@ def test_create_body_from_loft_profile(modeler: Modeler):
...
+    same origin."""
+    # Initialize the donut sketch design
+    design = modeler.create_design("revolve-fail")
+
+    # Create an XZ-plane centered at (0, 0, 0)
```
Comment (PipKat) [inline tests/integration/test_design.py:2327]:
```suggestion
    # Create an XZ plane centered at (0, 0, 0)
```

File: tests/integration/test_plotter.py [hunk 20]
```diff
@@ -688,3 +690,73 @@ def test_plot_design_point(modeler: Modeler, verify_image_cache):
...
+def test_plot_revolve_sketch_normal(modeler: Modeler):
+    """Test plotting of a sketch revolved around an axis."""
+    # Initialize the donut sketch design
+    design = modeler.create_design("quarter-donut")
+
+    # Donut parameters
+    path_radius = 5
+    profile_radius = 2
+
+    # Create the circular profile on the XZ-plane centered at (5, 0, 0)
```
Comment (PipKat) [inline tests/integration/test_plotter.py:704]:
```suggestion
    # Create the circular profile on the XZ plane centered at (5, 0, 0)
```
This wording is probably better than the "and centered" wording. Perhaps search and replace to make consistent?

File: tests/integration/test_plotter.py [hunk 21]
```diff
@@ -688,3 +690,73 @@ def test_plot_design_point(modeler: Modeler, verify_image_cache):
...
+    # with a radius of 2
+    plane_profile = Plane(
+        origin=Point3D([path_radius, 0, 0]), direction_x=UNITVECTOR3D_X, direction_y=UNITVECTOR3D_Z
+    )
+    profile = Sketch(plane=plane_profile)
+    profile.circle(Point2D([0, 0]), profile_radius)
+
+    # Revolve the profile around the Z-axis and centered in the absolute origin
+    # for an angle of 90 degrees
```
Comment (PipKat) [inline tests/integration/test_plotter.py:713]:
```suggestion
    # Revolve the profile around the Z axis and center in the absolute origin
    # for an angle of 90 degrees
```

File: tests/integration/test_plotter.py [hunk 22]
```diff
@@ -688,3 +690,73 @@ def test_plot_design_point(modeler: Modeler, verify_image_cache):
...
+        screenshot=Path(IMAGE_RESULTS_DIR, "test_plot_revolve_sketch_normal.png"),
+    )
+
+
+def test_plot_revolve_sketch_negative_angle(modeler: Modeler):
+    """Test plotting of a sketch revolved around an axis with a negative angle."""
+    # Initialize the donut sketch design
+    design = modeler.create_design("quarter-donut")
+
+    # Donut parameters
```
Comment (PipKat) [inline tests/integration/test_plotter.py:735]:
```suggestion
    # Specify donut parameters
```

File: tests/integration/test_plotter.py [hunk 23]
```diff
@@ -688,3 +690,73 @@ def test_plot_design_point(modeler: Modeler, verify_image_cache):
...
+    path_radius = 5
+    profile_radius = 2
+
+    # Create the circular profile on the XZ-plane centered at (5, 0, 0)
```
Comment (PipKat) [inline tests/integration/test_plotter.py:739]:
```suggestion
    # Create the circular profile on the XZ plane centered at (5, 0, 0)
```

File: tests/integration/test_plotter.py [hunk 24]
```diff
@@ -688,3 +690,73 @@ def test_plot_design_point(modeler: Modeler, verify_image_cache):
...
+    # with a radius of 2
+    plane_profile = Plane(
+        origin=Point3D([path_radius, 0, 0]), direction_x=UNITVECTOR3D_X, direction_y=UNITVECTOR3D_Z
+    )
+    profile = Sketch(plane=plane_profile)
+    profile.circle(Point2D([0, 0]), profile_radius)
+
+    # Revolve the profile around the Z-axis and centered in the absolute origin
```
Comment (PipKat) [inline tests/integration/test_plotter.py:747]:
```suggestion
    # Revolve the profile around the Z axis and centered in the absolute origin
```

File: tests/integration/test_design.py (hunk 18 above)
Comment (RobPasMue) [reply tests/integration/test_design.py:2322]: Ah it was mostly for the sketch - let me rephrase that

Comment (RobPasMue) [reply tests/integration/test_design.py:2322]:
```suggestion
    """Test demonstrating the failure of revolving a sketch when it is located in the
```
